src/field/gen_fieldops.py

```diff
@@ -0,0 +1,211 @@
+#!/usr/bin/env python3
+"""Code-generator for arithmetic operators on Field2Ds/Field3Ds
+
+This uses the jinja template in gen_fieldops.jinja to generate code
+for the arithmetic operators, and prints to stdout.
+
+The `Field` class provides some helper functions for determining how to
+pass a variable by reference or pointer, and how to name arguments in
+function signatures. This allows us to push some logic into the
+templates themselves.
+
+"""
+
+
+from __future__ import print_function
+
+try:
+    from builtins import object
+except ImportError:
+    pass
+
+from collections import OrderedDict
+from copy import deepcopy as copy
+import itertools
+
+try:
+    import jinja2
+except ImportError:
+    raise ImportError('Missing Python module "jinja2". See "Field2D/Field3D Arithmetic '
+                      'Operators" in the BOUT++ user manual for more information')
+
+# The arthimetic operators
+# OrderedDict to (try to) ensure consistency between python 2 & 3
```

Comment:
Presumably order just matters to prevent issues with committing the generated code and this triggering diffs even when the routines aren't changed (just the order)?

Reply:
Yep, exactly